--- a/com/google/javascript/jscomp/NodeTraversal.java
+++ b/com/google/javascript/jscomp/NodeTraversal.java
@@ -538,7 +538,7 @@ private void traverseFunction(Node n, Node parent) {
 
     // Body
     Preconditions.checkState(body.getNext() == null &&
-            body.isBlock());
+            body.isBlock(), body);
     traverseBranch(body, n);
 
     popScope();
diff --git a/com/google/javascript/jscomp/parsing/IRFactory.java b/com/google/javascript/jscomp/parsing/IRFactory.java
index 4da4085..26cbc19 100644
--- a/com/google/javascript/jscomp/parsing/IRFactory.java
+++ b/com/google/javascript/jscomp/parsing/IRFactory.java
@@ -664,9 +664,13 @@ Node processFunctionNode(FunctionNode functionNode) {
       node.addChildToBack(lp);
 
       Node bodyNode = transform(functionNode.getBody());
+      if (!bodyNode.isBlock()) {
         // When in ideMode Rhino tries to parse some constructs the compiler
         // doesn't support, repair it here. see Rhino's
         // Parser#parseFunctionBodyExpr.
+        Preconditions.checkState(config.isIdeMode);
+        bodyNode = IR.block();
+      }
       parseDirectives(bodyNode);
       node.addChildToBack(bodyNode);
      return node;
