--- a/org/jfree/data/time/TimeSeries.java
+++ b/org/jfree/data/time/TimeSeries.java
@@ -1054,6 +1054,8 @@ public TimeSeries createCopy(int start, int end)
             throw new IllegalArgumentException("Requires start <= end.");
         }
         TimeSeries copy = (TimeSeries) super.clone();
+        copy.minY = Double.NaN;
+        copy.maxY = Double.NaN;
         copy.data = new java.util.ArrayList();
         if (this.data.size() > 0) {
             for (int index = start; index <= end; index++) {
