--- a/core/src/main/java/org/apache/accumulo/core/security/crypto/BlockedOutputStream.java
+++ b/core/src/main/java/org/apache/accumulo/core/security/crypto/BlockedOutputStream.java
@@ -72,15 +72,18 @@ public class BlockedOutputStream extends OutputStream {
 
   @Override
   public void write(byte b[], int off, int len) throws IOException {
-    if (bb.remaining() >= len) {
-      bb.put(b, off, len);
-      if (bb.remaining() == 0)
-        flush();
-    } else {
+    // Can't recurse here in case the len is large and the blocksize is small (and the stack is small)
+    // So we'll just fill up the buffer over and over
+    while (len >= bb.remaining()) {
       int remaining = bb.remaining();
-      write(b, off, remaining);
-      write(b, off + remaining, len - remaining);
+      bb.put(b, off, remaining);
+      // This is guaranteed to have the buffer filled, so we'll just flush it. No check needed
+      flush();
+      off += remaining;
+      len -= remaining;
     }
+    // And then write the remainder (and this is guaranteed to not fill the buffer, so we won't flush afteward
+    bb.put(b, off, len);
   }
 
   @Override
