--- a/org/apache/commons/lang/text/StrBuilder.java
+++ b/org/apache/commons/lang/text/StrBuilder.java
@@ -1670,7 +1670,7 @@ public String midString(int index, int length) {
      */
     public boolean contains(char ch) {
         char[] thisBuf = buffer;
-        for (int i = 0; i < thisBuf.length; i++) {
+        for (int i = 0; i < this.size; i++) {
             if (thisBuf[i] == ch) {
                 return true;
             }
@@ -1727,7 +1727,7 @@ public int indexOf(char ch, int startIndex) {
             return -1;
         }
         char[] thisBuf = buffer;
-        for (int i = startIndex; i < thisBuf.length; i++) {
+        for (int i = startIndex; i < size; i++) {
             if (thisBuf[i] == ch) {
                 return i;
             }
