--- a/com/google/javascript/jscomp/Compiler.java
+++ b/com/google/javascript/jscomp/Compiler.java
@@ -252,7 +252,8 @@ public void initOptions(CompilerOptions options) {
           CheckLevel.OFF);
     }
 
-    if (options.checkGlobalThisLevel.isOn()) {
+    if (options.checkGlobalThisLevel.isOn() &&
+        !options.disables(DiagnosticGroups.GLOBAL_THIS)) {
       options.setWarningLevel(
           DiagnosticGroups.GLOBAL_THIS,
           options.checkGlobalThisLevel);
