--- a/com/google/javascript/rhino/jstype/NamedType.java
+++ b/com/google/javascript/rhino/jstype/NamedType.java
@@ -187,7 +187,7 @@ JSType resolveInternal(ErrorReporter t, StaticScope<JSType> enclosing) {
     // makes more sense. Now, resolution via registry is first in order to
     // avoid triggering the warnings built into the resolution via properties.
     boolean resolved = resolveViaRegistry(t, enclosing);
-    if (detectImplicitPrototypeCycle()) {
+    if (detectInheritanceCycle()) {
       handleTypeCycle(t);
     }
 
@@ -199,7 +199,7 @@ JSType resolveInternal(ErrorReporter t, StaticScope<JSType> enclosing) {
     }
 
     resolveViaProperties(t, enclosing);
-    if (detectImplicitPrototypeCycle()) {
+    if (detectInheritanceCycle()) {
       handleTypeCycle(t);
     }
 
