--- a/org/apache/commons/math/util/MathUtils.java
+++ b/org/apache/commons/math/util/MathUtils.java
@@ -409,7 +409,7 @@ public static double factorialLog(final int n) {
      * @since 1.1
      */
     public static int gcd(int u, int v) {
-        if (u * v == 0) {
+        if ((u == 0) || (v == 0)) {
             return (Math.abs(u) + Math.abs(v));
         }
         // keep u and v negative, as negative integers range down to
