diff --git a/src/main/java/org/joda/time/LocalDate.java b/src/main/java/org/joda/time/LocalDate.java
index 3868531..c86b9d7 100644
--- a/src/main/java/org/joda/time/LocalDate.java
+++ b/src/main/java/org/joda/time/LocalDate.java
@@ -209,0 +210 @@ public final class LocalDate
+        int era = calendar.get(Calendar.ERA);
@@ -212 +213 @@ public final class LocalDate
-            yearOfEra,
+            (era == GregorianCalendar.AD ? yearOfEra : 1 - yearOfEra),
@@ -242,0 +244 @@ public final class LocalDate
+        if (date.getTime() < 0) {
@@ -243,0 +246,4 @@ public final class LocalDate
+            GregorianCalendar cal = new GregorianCalendar();
+            cal.setTime(date);
+            return fromCalendarFields(cal);
+        }
diff --git a/src/main/java/org/joda/time/LocalDateTime.java b/src/main/java/org/joda/time/LocalDateTime.java
index f37db3e..e75deca 100644
--- a/src/main/java/org/joda/time/LocalDateTime.java
+++ b/src/main/java/org/joda/time/LocalDateTime.java
@@ -198,0 +199 @@ public final class LocalDateTime
+        int era = calendar.get(Calendar.ERA);
@@ -201 +202 @@ public final class LocalDateTime
-            yearOfEra,
+            (era == GregorianCalendar.AD ? yearOfEra : 1 - yearOfEra),
@@ -235,0 +237 @@ public final class LocalDateTime
+        if (date.getTime() < 0) {
@@ -236,0 +239,4 @@ public final class LocalDateTime
+            GregorianCalendar cal = new GregorianCalendar();
+            cal.setTime(date);
+            return fromCalendarFields(cal);
+        }
