--- a/com/google/javascript/jscomp/PeepholeFoldConstants.java
+++ b/com/google/javascript/jscomp/PeepholeFoldConstants.java
@@ -708,14 +708,12 @@ private Node performArithmeticOp(int opType, Node left, Node right) {
         break;
       case Token.MOD:
         if (rval == 0) {
-          error(DiagnosticType.error("JSC_DIVIDE_BY_0_ERROR", "Divide by 0"), right);
           return null;
         }
         result = lval % rval;
         break;
       case Token.DIV:
         if (rval == 0) {
-          error(DiagnosticType.error("JSC_DIVIDE_BY_0_ERROR", "Divide by 0"), right);
           return null;
         }
         result = lval / rval;
