--- a/org/apache/commons/lang/builder/EqualsBuilder.java
+++ b/org/apache/commons/lang/builder/EqualsBuilder.java
@@ -377,8 +377,12 @@ public EqualsBuilder append(Object lhs, Object rhs) {
         }
         Class lhsClass = lhs.getClass();
         if (!lhsClass.isArray()) {
+            if (lhs instanceof java.math.BigDecimal) {
+                isEquals = (((java.math.BigDecimal)lhs).compareTo(rhs) == 0);
+            } else {
                 // The simple case, not an array, just test the element
                 isEquals = lhs.equals(rhs);
+            }
         } else if (lhs.getClass() != rhs.getClass()) {
             // Here when we compare different dimensions, for example: a boolean[][] to a boolean[] 
             this.setEquals(false);
