--- a/com/google/javascript/jscomp/parsing/IRFactory.java
+++ b/com/google/javascript/jscomp/parsing/IRFactory.java
@@ -514,6 +514,13 @@ Node processFunctionNode(FunctionNode functionNode) {
       Name name = functionNode.getFunctionName();
       Boolean isUnnamedFunction = false;
       if (name == null) {
+        int functionType = functionNode.getFunctionType();
+        if (functionType != FunctionNode.FUNCTION_EXPRESSION) {
+          errorReporter.error(
+            "unnamed function statement",
+            sourceName,
+            functionNode.getLineno(), "", 0);
+        }
         name = new Name();
         name.setIdentifier("");
         isUnnamedFunction = true;
