--- a/com/google/javascript/jscomp/LiveVariablesAnalysis.java
+++ b/com/google/javascript/jscomp/LiveVariablesAnalysis.java
@@ -203,8 +203,12 @@ private void computeGenKill(Node n, BitSet gen, BitSet kill,
             // for(var x in y) {...}
             lhs = lhs.getLastChild();
           }
+          if (NodeUtil.isName(lhs)) {
             addToSetIfLocal(lhs, kill);
             addToSetIfLocal(lhs, gen);
+          } else {
+            computeGenKill(lhs, gen, kill, conditional);
+          }
           computeGenKill(rhs, gen, kill, conditional);
         }
         return;
