--- a/com/google/javascript/jscomp/GlobalNamespace.java
+++ b/com/google/javascript/jscomp/GlobalNamespace.java
@@ -904,6 +904,10 @@ boolean canCollapseUnannotatedChildNames() {
       // Don't try to collapse if the one global set is a twin reference.
       // We could theoretically handle this case in CollapseProperties, but
       // it's probably not worth the effort.
+      Preconditions.checkNotNull(declaration);
+      if (declaration.getTwin() != null) {
+        return false;
+      }
 
       if (isClassOrEnum) {
         return true;
diff --git a/com/google/javascript/rhino/JSDocInfoBuilder.java b/com/google/javascript/rhino/JSDocInfoBuilder.java
index 1311638..bdfc2e8 100644
--- a/com/google/javascript/rhino/JSDocInfoBuilder.java
+++ b/com/google/javascript/rhino/JSDocInfoBuilder.java
@@ -186,9 +186,7 @@ public void markName(String name, int lineno, int charno) {
    * @return {@code true} if the description was recorded.
    */
   public boolean recordBlockDescription(String description) {
-    if (parseDocumentation) {
     populated = true;
-    }
     return currentInfo.documentBlock(description);
   }
 
