--- a/org/jfree/chart/util/ShapeList.java
+++ b/org/jfree/chart/util/ShapeList.java
@@ -108,7 +108,14 @@ public boolean equals(Object obj) {
         if (!(obj instanceof ShapeList)) {
             return false;
         }
-        return super.equals(obj);
+        ShapeList that = (ShapeList) obj;
+        int listSize = size();
+        for (int i = 0; i < listSize; i++) {
+           if (!ShapeUtilities.equal((Shape) get(i), (Shape) that.get(i))) {
+               return false;
+           }
+        }
+        return true;
 
     }
 
