--- a/org/mockito/internal/configuration/injection/PropertyAndSetterInjection.java
+++ b/org/mockito/internal/configuration/injection/PropertyAndSetterInjection.java
@@ -111,7 +111,7 @@ private boolean injectMockCandidates(Class<?> awaitingInjectionClazz, Set<Object
     private boolean injectMockCandidatesOnFields(Set<Object> mocks, Object instance, boolean injectionOccurred, List<Field> orderedInstanceFields) {
         for (Iterator<Field> it = orderedInstanceFields.iterator(); it.hasNext(); ) {
             Field field = it.next();
-            Object injected = mockCandidateFilter.filterCandidate(mocks, field, instance).thenInject();
+            Object injected = mockCandidateFilter.filterCandidate(mocks, field, orderedInstanceFields, instance).thenInject();
             if (injected != null) {
                 injectionOccurred |= true;
                 mocks.remove(injected);
diff --git a/org/mockito/internal/configuration/injection/filter/FinalMockCandidateFilter.java b/org/mockito/internal/configuration/injection/filter/FinalMockCandidateFilter.java
index 7bd7073..aebf6f7 100644
--- a/org/mockito/internal/configuration/injection/filter/FinalMockCandidateFilter.java
+++ b/org/mockito/internal/configuration/injection/filter/FinalMockCandidateFilter.java
@@ -10,6 +10,7 @@
 
 import java.lang.reflect.Field;
 import java.util.Collection;
+import java.util.List;
 
 /**
  * This node returns an actual injecter which will be either :
@@ -20,7 +21,7 @@
  * </ul>
  */
 public class FinalMockCandidateFilter implements MockCandidateFilter {
-    public OngoingInjecter filterCandidate(final Collection<Object> mocks, final Field field, final Object fieldInstance) {
+    public OngoingInjecter filterCandidate(final Collection<Object> mocks, final Field field, List<Field> fields, final Object fieldInstance) {
         if(mocks.size() == 1) {
             final Object matchingMock = mocks.iterator().next();
 
diff --git a/org/mockito/internal/configuration/injection/filter/MockCandidateFilter.java b/org/mockito/internal/configuration/injection/filter/MockCandidateFilter.java
index 7a2c087..81a168b 100644
--- a/org/mockito/internal/configuration/injection/filter/MockCandidateFilter.java
+++ b/org/mockito/internal/configuration/injection/filter/MockCandidateFilter.java
@@ -6,13 +6,14 @@
 
 import java.lang.reflect.Field;
 import java.util.Collection;
+import java.util.List;
 
 public interface MockCandidateFilter {
 
     OngoingInjecter filterCandidate(
             Collection<Object> mocks,
             Field fieldToBeInjected,
-            Object fieldInstance
+            List<Field> fields, Object instance
     );
 
 }
diff --git a/org/mockito/internal/configuration/injection/filter/NameBasedCandidateFilter.java b/org/mockito/internal/configuration/injection/filter/NameBasedCandidateFilter.java
index b95f31a..abc5d67 100644
--- a/org/mockito/internal/configuration/injection/filter/NameBasedCandidateFilter.java
+++ b/org/mockito/internal/configuration/injection/filter/NameBasedCandidateFilter.java
@@ -20,7 +20,7 @@ public NameBasedCandidateFilter(MockCandidateFilter next) {
 	}
 
 	public OngoingInjecter filterCandidate(Collection<Object> mocks,
-			Field field, Object fieldInstance) {
+			Field field, List<Field> fields, Object fieldInstance) {
 		List<Object> mockNameMatches = new ArrayList<Object>();
 		if (mocks.size() > 1) {
 			for (Object mock : mocks) {
@@ -28,7 +28,7 @@ public OngoingInjecter filterCandidate(Collection<Object> mocks,
 					mockNameMatches.add(mock);
 				}
 			}
-			return next.filterCandidate(mockNameMatches, field,
+			return next.filterCandidate(mockNameMatches, field, fields,
 					fieldInstance);
 			/*
 			 * In this case we have to check whether we have conflicting naming
@@ -39,7 +39,23 @@ public OngoingInjecter filterCandidate(Collection<Object> mocks,
 			 * whenever we find a field that does match its name with the mock
 			 * name, we should take that field instead.
 			 */
+		} else if (mocks.size() == 1) {
+			String mockName = mockUtil.getMockName(mocks.iterator().next())
+					.toString();
+
+			for (Field otherField : fields) {
+				if (!otherField.equals(field)
+						&& otherField.getType().equals(field.getType())
+						&& otherField.getName().equals(mockName)) {
+
+					return new OngoingInjecter() {
+						public Object thenInject() {
+							return null;
+						}
+					};
+				}
+			}
 		}
-		return next.filterCandidate(mocks, field, fieldInstance);
+		return next.filterCandidate(mocks, field, fields, fieldInstance);
 	}
 }
diff --git a/org/mockito/internal/configuration/injection/filter/TypeBasedCandidateFilter.java b/org/mockito/internal/configuration/injection/filter/TypeBasedCandidateFilter.java
index dd79991..ce7a349 100644
--- a/org/mockito/internal/configuration/injection/filter/TypeBasedCandidateFilter.java
+++ b/org/mockito/internal/configuration/injection/filter/TypeBasedCandidateFilter.java
@@ -17,7 +17,7 @@ public TypeBasedCandidateFilter(MockCandidateFilter next) {
         this.next = next;
     }
 
-    public OngoingInjecter filterCandidate(Collection<Object> mocks, Field field, Object fieldInstance) {
+    public OngoingInjecter filterCandidate(Collection<Object> mocks, Field field, List<Field> fields, Object fieldInstance) {
         List<Object> mockTypeMatches = new ArrayList<Object>();
         for (Object mock : mocks) {
             if (field.getType().isAssignableFrom(mock.getClass())) {
@@ -25,6 +25,6 @@ public OngoingInjecter filterCandidate(Collection<Object> mocks, Field field, Ob
             }
         }
 
-        return next.filterCandidate(mockTypeMatches, field, fieldInstance);
+        return next.filterCandidate(mockTypeMatches, field, fields, fieldInstance);
     }
 }
