--- a/org/apache/commons/math3/util/MathArrays.java
+++ b/org/apache/commons/math3/util/MathArrays.java
@@ -818,7 +818,10 @@ public static double linearCombination(final double[] a, final double[] b)
             throw new DimensionMismatchException(len, b.length);
         }
 
+        if (len == 1) {
             // Revert to scalar multiplication.
+            return a[0] * b[0];
+        }
 
         final double[] prodHigh = new double[len];
         double prodLowSum = 0;
