--- a/core/src/main/java/dk/alexandra/fresco/framework/util/StrictBitVector.java
+++ b/core/src/main/java/dk/alexandra/fresco/framework/util/StrictBitVector.java
@@ -114,7 +114,8 @@ public class StrictBitVector implements BitVector {
    */
   @Override
   public byte[] toByteArray() {
-    return bits;
+    // TODO when we return bits directly we run into correlation errors in TestParallelMascots.
+    return bits.clone();
   }
 
   /**
@@ -215,4 +216,4 @@ public class StrictBitVector implements BitVector {
       throw new IndexOutOfBoundsException("Index out of bounds");
     }
   }
-}
\ No newline at end of file
+}
