--- a/org/jfree/chart/plot/XYPlot.java
+++ b/org/jfree/chart/plot/XYPlot.java
@@ -4490,6 +4490,7 @@ public Range getDataRange(ValueAxis axis) {
                     }
                 }
                 
+                if (r != null) {
                     Collection c = r.getAnnotations();
                     Iterator i = c.iterator();
                     while (i.hasNext()) {
@@ -4498,6 +4499,7 @@ public Range getDataRange(ValueAxis axis) {
                             includedAnnotations.add(a);
                         }
                     }
+                }
             }
         }
 
