--- a/org/mockito/exceptions/Reporter.java
+++ b/org/mockito/exceptions/Reporter.java
@@ -435,10 +435,11 @@ public void misplacedArgumentMatcher(Location location) {
                 ));
     }
 
-    public void smartNullPointerException(Location location) {
+    public void smartNullPointerException(Object obj, Location location) {
         throw new SmartNullPointerException(join(
                 "You have a NullPointerException here:",
                 new Location(),
+                obj,
                 "Because this method was *not* stubbed correctly:",
                 location,
                 ""
diff --git a/org/mockito/internal/stubbing/defaultanswers/ReturnsSmartNulls.java b/org/mockito/internal/stubbing/defaultanswers/ReturnsSmartNulls.java
index 4942d7c..928f314 100644
--- a/org/mockito/internal/stubbing/defaultanswers/ReturnsSmartNulls.java
+++ b/org/mockito/internal/stubbing/defaultanswers/ReturnsSmartNulls.java
@@ -53,7 +53,7 @@ public Object intercept(Object obj, Method method, Object[] args, MethodProxy pr
                 return "SmartNull returned by unstubbed " + formatMethodCall()  + " method on mock";
             }
 
-            new Reporter().smartNullPointerException(location);
+            new Reporter().smartNullPointerException(obj, location);
             return null;
         }
 
