--- a/com/google/javascript/jscomp/TypeCheck.java
+++ b/com/google/javascript/jscomp/TypeCheck.java
@@ -513,7 +513,9 @@ public void visit(NodeTraversal t, Node n, Node parent) {
         // Object literal keys are handled with OBJECTLIT
         if (!NodeUtil.isObjectLitKey(n, n.getParent())) {
           ensureTyped(t, n, STRING_TYPE);
+        } else {
           // Object literal keys are not typeable
+          typeable = false;
         }
         break;
 
