--- a/com/google/javascript/jscomp/GlobalNamespace.java
+++ b/com/google/javascript/jscomp/GlobalNamespace.java
@@ -363,6 +363,7 @@ public void collect(JSModule module, Scope scope, Node n) {
                 isSet = true;
                 type = Name.Type.FUNCTION;
                 break;
+              case Token.CATCH:
               case Token.INC:
               case Token.DEC:
                 isSet = true;
