--- a/com/google/javascript/jscomp/CommandLineRunner.java
+++ b/com/google/javascript/jscomp/CommandLineRunner.java
@@ -859,6 +859,7 @@ protected CompilerOptions createOptions() {
       // so we might as well inline it. But shut off the i18n warnings,
       // because the user didn't really ask for i18n.
       options.messageBundle = new EmptyMessageBundle();
+      options.setWarningLevel(JsMessageVisitor.MSG_CONVENTIONS, CheckLevel.OFF);
     }
 
     return options;
