--- a/org/apache/commons/lang/Entities.java
+++ b/org/apache/commons/lang/Entities.java
@@ -847,6 +847,9 @@ public String unescape(String str) {
                             } else {
                                 entityValue = Integer.parseInt(entityName.substring(1));
                             }
+                            if (entityValue > 0xFFFF) {
+                                entityValue = -1;
+                            }
                         } catch (NumberFormatException ex) {
                             entityValue = -1;
                         }
@@ -917,12 +920,17 @@ public void unescape(Writer writer, String string) throws IOException {
                                     case 'X' :
                                     case 'x' : {
                                         entityValue = Integer.parseInt(entityContent.substring(2), 16);
+                                        break;
                                     }
                                     default : {
                                         entityValue = Integer.parseInt(entityContent.substring(1), 10);
                                     }
                                 }
+                                if (entityValue > 0xFFFF) {
+                                    entityValue = -1;
+                                }
                             } catch (NumberFormatException e) {
+                                entityValue = -1;
                             }
                         }
                     } else { //escaped value content is an entity name
