--- a/org/mockito/Matchers.java
+++ b/org/mockito/Matchers.java
@@ -119,7 +119,7 @@
      * @return <code>false</code>.
      */
     public static boolean anyBoolean() {
-        return reportMatcher(Any.ANY).returnFalse();
+        return reportMatcher(new InstanceOf(Boolean.class)).returnFalse();
     }
 
     /**
@@ -134,7 +134,7 @@ public static boolean anyBoolean() {
      * @return <code>0</code>.
      */
     public static byte anyByte() {
-        return reportMatcher(Any.ANY).returnZero();
+        return reportMatcher(new InstanceOf(Byte.class)).returnZero();
     }
 
     /**
@@ -149,7 +149,7 @@ public static byte anyByte() {
      * @return <code>0</code>.
      */
     public static char anyChar() {
-        return reportMatcher(Any.ANY).returnChar();
+        return reportMatcher(new InstanceOf(Character.class)).returnChar();
     }
 
     /**
@@ -164,7 +164,7 @@ public static char anyChar() {
      * @return <code>0</code>.
      */
     public static int anyInt() {
-        return reportMatcher(Any.ANY).returnZero();
+        return reportMatcher(new InstanceOf(Integer.class)).returnZero();
     }
 
     /**
@@ -179,7 +179,7 @@ public static int anyInt() {
      * @return <code>0</code>.
      */
     public static long anyLong() {
-        return reportMatcher(Any.ANY).returnZero();
+        return reportMatcher(new InstanceOf(Long.class)).returnZero();
     }
 
     /**
@@ -194,7 +194,7 @@ public static long anyLong() {
      * @return <code>0</code>.
      */
     public static float anyFloat() {
-        return reportMatcher(Any.ANY).returnZero();
+        return reportMatcher(new InstanceOf(Float.class)).returnZero();
     }
 
     /**
@@ -209,7 +209,7 @@ public static float anyFloat() {
      * @return <code>0</code>.
      */
     public static double anyDouble() {
-        return reportMatcher(Any.ANY).returnZero();
+        return reportMatcher(new InstanceOf(Double.class)).returnZero();
     }
 
     /**
@@ -224,7 +224,7 @@ public static double anyDouble() {
      * @return <code>0</code>.
      */
     public static short anyShort() {
-        return reportMatcher(Any.ANY).returnZero();
+        return reportMatcher(new InstanceOf(Short.class)).returnZero();
     }
 
     /**
@@ -241,7 +241,7 @@ public static short anyShort() {
      * @return <code>null</code>.
      */
     public static <T> T anyObject() {
-        return (T) reportMatcher(Any.ANY).returnNull();
+        return (T) reportMatcher(new InstanceOf(Object.class)).returnNull();
     }
 
     /**
@@ -289,7 +289,7 @@ public static short anyShort() {
      * @return <code>null</code>.
      */
     public static <T> T any(Class<T> clazz) {
-        return (T) reportMatcher(Any.ANY).returnFor(clazz);
+        return (T) reportMatcher(new InstanceOf(clazz)).returnFor(clazz);
     }
     
     /**
@@ -306,7 +306,7 @@ public static short anyShort() {
      * @return <code>null</code>.
      */
     public static <T> T any() {
-        return (T) anyObject();
+        return (T) reportMatcher(Any.ANY).returnNull();
     }
 
     /**
@@ -321,7 +321,7 @@ public static short anyShort() {
      * @return empty String ("")
      */
     public static String anyString() {
-        return reportMatcher(Any.ANY).returnString();
+        return reportMatcher(new InstanceOf(String.class)).returnString();
     }
     
     /**
@@ -336,7 +336,7 @@ public static String anyString() {
      * @return empty List.
      */
     public static List anyList() {
-        return reportMatcher(Any.ANY).returnList();
+        return reportMatcher(new InstanceOf(List.class)).returnList();
     }    
     
     /**
@@ -355,7 +355,7 @@ public static List anyList() {
      * @return empty List.
      */
     public static <T> List<T> anyListOf(Class<T> clazz) {
-        return (List) reportMatcher(Any.ANY).returnList();
+        return anyList();
     }    
     
     /**
@@ -370,7 +370,7 @@ public static List anyList() {
      * @return empty Set
      */
     public static Set anySet() {
-        return reportMatcher(Any.ANY).returnSet();
+        return reportMatcher(new InstanceOf(Set.class)).returnSet();
     }
     
     /**
@@ -389,7 +389,7 @@ public static Set anySet() {
      * @return empty Set
      */
     public static <T> Set<T> anySetOf(Class<T> clazz) {
-        return (Set) reportMatcher(Any.ANY).returnSet();
+        return anySet();
     }
 
     /**
@@ -404,7 +404,7 @@ public static Set anySet() {
      * @return empty Map.
      */
     public static Map anyMap() {
-        return reportMatcher(Any.ANY).returnMap();
+        return reportMatcher(new InstanceOf(Map.class)).returnMap();
     }
 
     /**
@@ -424,7 +424,7 @@ public static Map anyMap() {
      * @return empty Map.
      */
     public static <K, V>  Map<K, V> anyMapOf(Class<K> keyClazz, Class<V> valueClazz) {
-        return reportMatcher(Any.ANY).returnMap();
+        return anyMap();
     }
     
     /**
@@ -439,7 +439,7 @@ public static Map anyMap() {
      * @return empty Collection.
      */
     public static Collection anyCollection() {
-        return reportMatcher(Any.ANY).returnList();
+        return reportMatcher(new InstanceOf(Collection.class)).returnList();
     }    
     
     /**
@@ -458,7 +458,7 @@ public static Collection anyCollection() {
      * @return empty Collection.
      */
     public static <T> Collection<T> anyCollectionOf(Class<T> clazz) {
-        return (Collection) reportMatcher(Any.ANY).returnList();
+        return anyCollection();
     }    
 
     /**
