--- a/org/apache/commons/math3/dfp/Dfp.java
+++ b/org/apache/commons/math3/dfp/Dfp.java
@@ -1600,7 +1600,11 @@ public Dfp multiply(final Dfp x) {
      * @return product of this and x
      */
     public Dfp multiply(final int x) {
+        if (x >= 0 && x < RADIX) {
             return multiplyFast(x);
+        } else {
+            return multiply(newInstance(x));
+        }
     }
 
     /** Multiply this by a single digit 0&lt;=x&lt;radix.
