--- a/org/mockito/internal/configuration/SpyAnnotationEngine.java
+++ b/org/mockito/internal/configuration/SpyAnnotationEngine.java
@@ -16,6 +16,7 @@
 import org.mockito.exceptions.base.MockitoException;
 import org.mockito.internal.util.MockUtil;
 
+import static org.mockito.Mockito.withSettings;
 
 @SuppressWarnings({"unchecked"})
 public class SpyAnnotationEngine implements AnnotationEngine {
@@ -46,7 +47,10 @@ public void process(Class<?> context, Object testClass) {
                         // instance has been spied earlier
                         Mockito.reset(instance);
                     } else {
-                        field.set(testClass, Mockito.spy(instance));
+                        field.set(testClass, Mockito.mock(instance.getClass(), withSettings()
+                                .spiedInstance(instance)
+                                .defaultAnswer(Mockito.CALLS_REAL_METHODS)
+                                .name(field.getName())));
                     }
                 } catch (IllegalAccessException e) {
                     throw new MockitoException("Problems initiating spied field " + field.getName(), e);
