diff --git a/mnt/secondary/projects/closure/closure_24/src/com/google/javascript/jscomp/ScopedAliases.java b/mnt/secondary/projects_fix/closure/closure_24/src/com/google/javascript/jscomp/ScopedAliases.java
index 30d29d0..c248002 100644
--- a/mnt/secondary/projects/closure/closure_24/src/com/google/javascript/jscomp/ScopedAliases.java
+++ b/mnt/secondary/projects_fix/closure/closure_24/src/com/google/javascript/jscomp/ScopedAliases.java
@@ -273,28 +273,29 @@ class ScopedAliases implements HotSwapCompilerPass {
       Scope scope = t.getScope();
       for (Var v : scope.getVarIterable()) {
         Node n = v.getNode();
         int type = n.getType();
         Node parent = n.getParent();
-        if (parent.isVar()) {
-          if (n.hasChildren() && n.getFirstChild().isQualifiedName()) {
+        if (parent.isVar() &&
+            n.hasChildren() && n.getFirstChild().isQualifiedName()) {
           String name = n.getString();
           Var aliasVar = scope.getVar(name);
           aliases.put(name, aliasVar);
 
           String qualifiedName =
               aliasVar.getInitialValue().getQualifiedName();
           transformation.addAlias(name, qualifiedName);
+        } else if (v.isBleedingFunction()) {
           // Bleeding functions already get a BAD_PARAMETERS error, so just
           // do nothing.
+        } else if (parent.getType() == Token.LP) {
           // Parameters of the scope function also get a BAD_PARAMETERS
           // error.
         } else {
           // TODO(robbyw): Support using locals for private variables.
           report(t, n, GOOG_SCOPE_NON_ALIAS_LOCAL, n.getString());
         }
-        }
       }
     }
 
     private void validateScopeCall(NodeTraversal t, Node n, Node parent) {
       if (preprocessorSymbolTable != null) {
