--- a/org/apache/commons/lang3/builder/ToStringStyle.java
+++ b/org/apache/commons/lang3/builder/ToStringStyle.java
@@ -145,7 +145,7 @@
      * @return Set the registry of objects being traversed
      */
     static Map<Object, Object> getRegistry() {
-        return REGISTRY.get() != null ? REGISTRY.get() : Collections.<Object, Object>emptyMap();
+        return REGISTRY.get();
     }
 
     /**
@@ -161,7 +161,7 @@
      */
     static boolean isRegistered(Object value) {
         Map<Object, Object> m = getRegistry();
-        return m.containsKey(value);
+        return m != null && m.containsKey(value);
     }
 
     /**
