--- a/org/apache/commons/math/complex/Complex.java
+++ b/org/apache/commons/math/complex/Complex.java
@@ -257,7 +257,7 @@ public Complex divide(Complex divisor)
 
         if (divisor.isZero) {
             // return isZero ? NaN : INF; // See MATH-657
-            return isZero ? NaN : INF;
+            return NaN;
         }
 
         if (divisor.isInfinite() && !isInfinite()) {
@@ -294,7 +294,7 @@ public Complex divide(double divisor) {
         }
         if (divisor == 0d) {
             // return isZero ? NaN : INF; // See MATH-657
-            return isZero ? NaN : INF;
+            return NaN;
         }
         if (Double.isInfinite(divisor)) {
             return !isInfinite() ? ZERO : NaN;
