--- a/com/google/javascript/jscomp/FunctionTypeBuilder.java
+++ b/com/google/javascript/jscomp/FunctionTypeBuilder.java
@@ -289,6 +289,9 @@ FunctionTypeBuilder inferFromOverriddenFunction(
       }
 
       // Clone any remaining params that aren't in the function literal.
+      while (oldParams.hasNext()) {
+        paramBuilder.newParameterFromNode(oldParams.next());
+      }
 
       parametersNode = paramBuilder.build();
     }
@@ -480,6 +483,12 @@ FunctionTypeBuilder inferParameterTypes(@Nullable Node argsParent,
     }
 
     // Copy over any old parameters that aren't in the param list.
+    if (!isVarArgs) {
+      while (oldParameterType != null && !isVarArgs) {
+        builder.newParameterFromNode(oldParameterType);
+        oldParameterType = oldParameterType.getNext();
+      }
+    }
 
     if (templateTypeName != null && !foundTemplateType) {
       reportError(TEMPLATE_TYPE_EXPECTED, fnName);
