--- a/com/google/javascript/jscomp/FlowSensitiveInlineVariables.java
+++ b/com/google/javascript/jscomp/FlowSensitiveInlineVariables.java
@@ -154,7 +154,7 @@ public void exitScope(NodeTraversal t) {}
 
   @Override
   public void process(Node externs, Node root) {
-    (new NodeTraversal(compiler, this)).traverse(root);
+    (new NodeTraversal(compiler, this)).traverseRoots(externs, root);
   }
 
   @Override
diff --git a/com/google/javascript/jscomp/MustBeReachingVariableDef.java b/com/google/javascript/jscomp/MustBeReachingVariableDef.java
index 1487af3..589f673 100644
--- a/com/google/javascript/jscomp/MustBeReachingVariableDef.java
+++ b/com/google/javascript/jscomp/MustBeReachingVariableDef.java
@@ -68,6 +68,7 @@
   private static class Definition {
     final Node node;
     final Set<Var> depends = Sets.newHashSet();
+    private boolean unknownDependencies = false;
 
     Definition(Node node) {
       this.node = node;
@@ -393,9 +394,13 @@ private void computeDependence(final Definition def, Node rValue) {
         new AbstractCfgNodeTraversalCallback() {
       @Override
       public void visit(NodeTraversal t, Node n, Node parent) {
-        if (n.isName() && jsScope.isDeclared(n.getString(), true)) {
+        if (n.isName()) {
           Var dep = jsScope.getVar(n.getString());
+          if (dep == null) {
+            def.unknownDependencies = true;
+          } else {
             def.depends.add(dep);
+          }
         }
       }
     });
@@ -427,6 +432,9 @@ boolean dependsOnOuterScopeVars(String name, Node useNode) {
     GraphNode<Node, Branch> n = getCfg().getNode(useNode);
     FlowState<MustDef> state = n.getAnnotation();
     Definition def = state.getIn().reachingDef.get(jsScope.getVar(name));
+    if (def.unknownDependencies) {
+      return true;
+    }
 
     for (Var s : def.depends) {
       if (s.scope != jsScope) {
