--- a/com/google/javascript/jscomp/InlineVariables.java
+++ b/com/google/javascript/jscomp/InlineVariables.java
@@ -572,6 +572,9 @@ private boolean canInline(
 
           // issue 668: Don't inline singleton getter methods
           // calls as this confused class removing logic.
+          if (convention.getSingletonGetterClassName(callNode) != null) {
+            return false;
+          }
         }
       }
 
