--- a/src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializer.java
+++ b/src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializer.java
@@ -373,6 +373,10 @@ public class BeanDeserializer
         for (; t == JsonToken.FIELD_NAME; t = p.nextToken()) {
             String propName = p.getCurrentName();
             p.nextToken(); // to point to value
+            // Object Id property?
+            if (buffer.readIdProperty(propName)) {
+                continue;
+            }
             // creator property?
             SettableBeanProperty creatorProp = creator.findCreatorProperty(propName);
             if (creatorProp != null) {
@@ -405,10 +409,6 @@ public class BeanDeserializer
                 }
                 continue;
             }
-            // Object Id property?
-            if (buffer.readIdProperty(propName)) {
-                continue;
-            }
             // regular property? needs buffering
             SettableBeanProperty prop = _beanProperties.find(propName);
             if (prop != null) {
