--- a/org/apache/commons/math/complex/ComplexFormat.java
+++ b/org/apache/commons/math/complex/ComplexFormat.java
@@ -374,7 +374,8 @@ public Complex parse(String source, ParsePosition pos) {
         int n = getImaginaryCharacter().length();
         startIndex = pos.getIndex();
         int endIndex = startIndex + n;
-        if (
+        if ((startIndex >= source.length()) ||
+            (endIndex > source.length()) ||
             source.substring(startIndex, endIndex).compareTo(
             getImaginaryCharacter()) != 0) {
             // set index back to initial, error index should be the start index
